### ISSUE TITLE
[bug/feat] Allow Bitcoin Core RPC / built-in bwt to accept https:// urls

### ISSUE THREAD

**Issue by gStart9** [sparrowwallet/sparrow#768]
I'd say this one falls somewhere between a bug and a feature request 😁

On embassyOS, we allow connections to the bitcoin rpc via an https address only so credentials cannot be intercepted on the network.  nginx proxies that connection to the bitcoin core container's rpc port.

In Sparrow 1.7.1 for debian amd64, when I go to `File > Preferences > Server > Bitcoin Core`
I enter https://mylocalbitcoinrpcnodeaddressname.local:443

Unfortunately, when I specify https:// , it strips the :// and just attempts to contact http://httpsmylocalbitcoinrpcnodeaddressname.local

Is it a heavy lift to feed an https address to bwt?

Next step - https://mylocalbitcoinrpcnodeaddressname.local uses a custom CA, which I've imported to my certificate store using:
```
sudo cp mycacert.crt /usr/loca/share/ca-certificates/
sudo update-ca-certificates
```

Any hurdles there?

**Comment by craigraw** [comment]
Personally I'm of the view that if you don't trust your local network, you probably shouldn't expose Bitcoin Core RPC to it at all. I don't think that exposing the RPC over untrusted networks should be encouraged (and I recommend again using it over Tor).

That said Sparrow should not try to be too prescriptive, so I'm happy to allow this usage if the port is explicitly set to 443. However BWT does not appear to support https calls at all, and additionally has no configuration for a self-signed certificate (using `ca-certificates` is not a cross-platform approach). I am however working on native Bitcoin Core support in Sparrow, and I'll try to include this use case.

**Comment by craigraw** [comment]
> That said Sparrow should not try to be too prescriptive, so I'm happy to allow this usage if the port is explicitly set to 443.

I've included this as part of 08cf01a5. Sparrow doesn't try to verify the server certificate, since the main point here is encryption and not verification.
